scipy/signal/_ltisys.py

```diff
@@ -3367,24 +3392,25 @@ def dfreqresp(system, w=None, n=10000, whole=False):
 
     Examples
     --------
-    Generating the Nyquist plot of a transfer function
+    The following example generates the Nyquist plot of the transfer function
+    :math:`H(z) = \frac{1}{z^2 + 2z + 3}`  with a sampling time of 0.05 seconds:
 
     >>> from scipy import signal
     >>> import matplotlib.pyplot as plt
-
-    Construct the transfer function
-    :math:`H(z) = \frac{1}{z^2 + 2z + 3}` with a sampling time of 0.05
-    seconds:
-
-    >>> sys = signal.TransferFunction([1], [1, 2, 3], dt=0.05)
-
+    >>> sys = signal.TransferFunction([1], [1, 2, 3], dt=0.05)  # construct H(z)
     >>> w, H = signal.dfreqresp(sys)
-
-    >>> plt.figure()
-    >>> plt.plot(H.real, H.imag, "b")
-    >>> plt.plot(H.real, -H.imag, "r")
+    ...
+    >>> fig0, ax0 = plt.subplots()
+    >>> ax0.plot(H.real, H.imag, label=r"$H(z=e^{+j\omega})$")
+    >>> ax0.plot(H.real, -H.imag, label=r"$H(z=e^{-j\omega})$")
+    >>> ax0.set_title(r"Nyquist Plot of $H(z) = 1 / (z^2 + 2z + 3)$")
+    >>> ax0.set(xlabel=r"$\text{Re}\{z\}$", ylabel=r"$\text{Im}\{z\}$",
+    ...         xlim=(-0.2, 0.65), aspect='equal')
+    >>> ax0.plot(H[0].real, H[0].imag, 'k.')  # mark H(exp(1j*w[0]))
+    >>> _ = ax0.text(0, 0.2, r"$H(e^{j0})$")  # `_ =` needed to pass doctest
```

Comment:
Possibly 
```suggestion
    >>> _ = ax0.text(0, 0, r"$H(e^{j0})$")  # `_ =` needed to pass doctest
```
As currently they don't line up too well 
![Screenshot_20250201-061916](https://github.com/user-attachments/assets/0dc10b7e-dea0-4035-af98-a46a4292d309)

Reply:
Also consider using this diff below instead of contorting the docstring for the doctester:

```diff
$ git diff
diff --git a/scipy/conftest.py b/scipy/conftest.py
index d7f737a2eb..784265b7e9 100644
--- a/scipy/conftest.py
+++ b/scipy/conftest.py
@@ -599,4 +599,7 @@ if HAVE_SCPDT:
     }
 
     dt_config.strict_check = True
+
+    # ignore Matplotlib's ax.text
+    dt_config.stopwords.add('.text(')
 ############################################################################
diff --git a/scipy/signal/_ltisys.py b/scipy/signal/_ltisys.py
index 4e88d9666f..e1ff1fde3c 100644
--- a/scipy/signal/_ltisys.py
+++ b/scipy/signal/_ltisys.py
@@ -3407,7 +3407,7 @@ def dfreqresp(system, w=None, n=10000, whole=False):
     >>> ax0.set(xlabel=r"$\text{Re}\{z\}$", ylabel=r"$\text{Im}\{z\}$",
     ...         xlim=(-0.2, 0.65), aspect='equal')
     >>> ax0.plot(H[0].real, H[0].imag, 'k.')  # mark H(exp(1j*w[0]))
-    >>> _ = ax0.text(0, 0.2, r"$H(e^{j0})$")  # `_ =` needed to pass doctest
+    >>> ax0.text(0, 0.2, r"$H(e^{j0})$")
     >>> ax0.grid(True)
     >>> ax0.legend()
     >>> plt.show()

```

Reply:
I was not aware of `scipy/conftest.py`. Thanks! 
I did not catch the wrong text position in the plot locally, since the plots are not rebuilt if the docstr changes. 

Fixed in commit https://github.com/scipy/scipy/pull/22453/commits/125e9b0f361c437af7aa01f644e5ab3e49dbc57f. The rendered  version of the docstrs can be found here:
* [dimpluse](https://output.circle-artifacts.com/output/job/94c9dd2d-f13d-491f-975c-ad99999d744e/artifacts/0/html/reference/generated/scipy.signal.dimpulse.html#scipy.signal.dimpulse)
* [dstep](https://output.circle-artifacts.com/output/job/94c9dd2d-f13d-491f-975c-ad99999d744e/artifacts/0/html/reference/generated/scipy.signal.dstep.html#scipy.signal.dstep)
* [dfeqresp](https://output.circle-artifacts.com/output/job/94c9dd2d-f13d-491f-975c-ad99999d744e/artifacts/0/html/reference/generated/scipy.signal.dfreqresp.html#scipy.signal.dfreqresp)
* [dbode](https://output.circle-artifacts.com/output/job/94c9dd2d-f13d-491f-975c-ad99999d744e/artifacts/0/html/reference/generated/scipy.signal.dbode.html#scipy.signal.dbode)